frontend_tests/node_tests/settings_bots.js

```diff
@@ -98,31 +98,100 @@ function test_create_bot_type_input_box_toggle(f) {
     assert(!config_inputbox.visible());
 }
 
-run_test('set_up', () => {
+function set_up() {
+    set_global('$', global.make_zjquery());
+
     // bunch of stubs
 
-    $.validator = { addMethod: function () {} };
+    $.validator = { addMethod: () => {} };
 
-    $("#create_bot_form").validate = function () {};
+    $("#create_bot_form").validate = () => {};
 
-    $('#create_bot_type').on = function (action, f) {
+    $('#create_bot_type').on = (action, f) => {
         if (action === 'change') {
             test_create_bot_type_input_box_toggle(f);
         }
     };
 
-    $('#config_inputbox').children = function () {
+    $('#config_inputbox').children = () => {
         var mock_children = {
-            hide: function () {
+            hide: () => {
                 return;
             },
         };
         return mock_children;
     };
     global.compile_template('embedded_bot_config_item');
-    avatar.build_bot_create_widget = function () {};
-    avatar.build_bot_edit_widget = function () {};
+    avatar.build_bot_create_widget = () => {};
+    avatar.build_bot_edit_widget = () => {};
 
     settings_bots.set_up();
+}
+
+run_test('set_up', () => {
+    set_up();
 });
 
+run_test('test tab clicks', () => {
+    set_up();
+
+    function click_on_tab(tab_elem) {
+        var e = {
+            preventDefault: () => {},
+            stopPropagation: () => {},
+        };
+        tab_elem.click(e);
+    }
+
+    const tabs = {
+        add: $('#bots_lists_navbar .add-a-new-bot-tab'),
+        active: $('#bots_lists_navbar .active-bots-tab'),
+        inactive: $('#bots_lists_navbar .inactive-bots-tab'),
+    };
+
+    $('#bots_lists_navbar .active').removeClass = (cls) => {
+        assert.equal(cls, 'active');
+        _.each(tabs, (tab) => {
+            tab.removeClass('active');
+        });
+    };
+
+    const forms = {
+        add: $('#add-a-new-bot-form'),
+        active: $('#active_bots_list'),
+        inactive: $('#inactive_bots_list'),
+    };
+
+    (function () {
+        click_on_tab(tabs.add);
+        assert(tabs.add.hasClass('active'));
+        assert(!tabs.active.hasClass('active'));
+        assert(!tabs.inactive.hasClass('active'));
+
+        assert(forms.add.visible());
+        assert(!forms.active.visible());
+        assert(!forms.inactive.visible());
+    }());
```

Comment:
@showell I want to know do we use these IIFEs for clarity or make to code inside it scoped; if later is the case then we can use another ES6 style here:
```js
{
    let food = 'Meow Mix';
};

console.log(food); // Reference Error
```
Though I'm not very strong about this.
Also, I can see some other instances of these IIFEs where we take advantage of the name of for clarity but we can use a comment like below 
```js
    function test_create_item(handler) {
        (function test_rejection_path() {
            var item = handler(othello.email, pills.items());
            assert(get_by_email_called);
            assert.equal(item, undefined);
        }());

        (function test_success_path() {
            get_by_email_called = false;
            var res = handler(iago.email, pills.items());
            assert(get_by_email_called);
            assert.equal(typeof res, 'object');
            assert.equal(res.user_id, iago.user_id);
            assert.equal(res.display_value, iago.full_name);
        }());
    }
```
can be written as:
```js
    function test_create_item(handler) {
        {
            /* Test rejection path */
            var item = handler(othello.email, pills.items());
            assert(get_by_email_called);
            assert.equal(item, undefined);
        }

        {
            /* Test success path */
            get_by_email_called = false;
            var res = handler(iago.email, pills.items());
            assert(get_by_email_called);
            assert.equal(typeof res, 'object');
            assert.equal(res.user_id, iago.user_id);
            assert.equal(res.display_value, iago.full_name);
        }
    }
```
Thoughts?

Reply:
Or there can be instances where this new ES6 doesn't make sense?

Reply:
Interesting!  I wasn't aware of the syntax.  Do curlies introduce a new scope or just provide syntax sugar to denote separate "blocks" of code?

Reply:
I think yes, they create a new scope, 
https://stackoverflow.com/a/33534601/7418550 This answer clears this.
https://wesbos.com/es6-block-scope-iife/

Reply:
Yep, they do provide new scope but i did bring them up because mostly i see them being use to write toplevel test like IIFE function. Node uses them for their own tests https://github.com/nodejs/node/blob/master/test/parallel/test-assert-async.js#L14-L36.

Reply:
But for this types of use cases it works well.